soh/soh/Enhancements/enemyrandomizer.cpp

```diff
@@ -0,0 +1,402 @@
+                    return (params == 0 || params == -32768);
+                // Only randomize the initial eggs, not the enemies that spawn from them.
+                case ACTOR_EN_GOMA:
+                    return (params >= 0 && params <= 9);
+                // Only randomize Skullwalltulas, not Golden Skulltulas.
+                case ACTOR_EN_SW:
+                    return (params == 0);
+                // Don't randomize Nabooru because it'll break the cutscene and the door.
+                // Don't randomize Iron Knuckle in MQ Spirit Trial because it's needed to
+                // break the thrones in the room to access a button.
+                case ACTOR_EN_IK:
+                    return (params != 1280 && !(isMQ && sceneNum == SCENE_GANONTIKA && roomNum == 17));
+                // Only randomize the intitial spawn of the huge jellyfish. It spawns another copy when hit with a sword.
+                case ACTOR_EN_VALI:
+                    return (params == -1);
+                // Don't randomize lizalfos in Doodong's Cavern because the gates won't work correctly otherwise.
+                case ACTOR_EN_ZF:
+                    return (params != 1280 && params != 1281 && params != 1536 && params != 1537);
+                // Don't randomize the Wolfos in SFM because it's needed to open the gate.
+                case ACTOR_EN_WF:
+                    return (params != 7936);
+                // Don't randomize the Stalfos in Forest Temple because other enemies fall through the hole and don't trigger the platform.
+                // Don't randomize the Stalfos spawning on the boat in Shadow Temple, as randomizing them places the new enemies
+                // down in the river.
+                case ACTOR_EN_TEST:
+                    return (params != 1 && !(sceneNum == SCENE_HAKADAN && roomNum == 21));
+                // Only randomize the enemy variant of Armos Statue.
+                // Leave one Armos unrandomized in the Spirit Temple room where an armos is needed to push down a button
+                case ACTOR_EN_AM:
+                    return ((params == -1 || params == 255) && !(sceneNum == SCENE_JYASINZOU && posX == 2141));
+                // Don't randomize Shell Blades and Spikes in the underwater portion in Water Temple as it's impossible to kill
+                // most other enemies underwater with just hookshot and they're required to be killed for a grate to open.
+                case ACTOR_EN_SB:
+                case ACTOR_EN_NY:
+                    return (!(!isMQ && sceneNum == SCENE_MIZUSIN && roomNum == 2));
+                default:
+                    return 1;
+            }
+        }
+    }
+
+    // If no enemy is found, don't randomize the actor.
+    return 0;
+}
+
+bool IsEnemyAllowedToSpawn(int16_t sceneNum, int8_t roomNum, EnemyEntry enemy) {
+
+    uint32_t isMQ = ResourceMgr_IsSceneMasterQuest(sceneNum);
+
+    // Freezard, Beamos, Shell Blade, Spike, Arwing, Wallmaster
```

Comment:
might be nice to have a little bit more of a explanation as to why these are being excluded